pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+
+	authenticationapiv1 "k8s.io/api/authentication/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	authenticationapi "k8s.io/kubernetes/pkg/apis/authentication"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	token "k8s.io/kubernetes/pkg/serviceaccount"
+)
+
+type TokenREST struct {
+	svcaccts getter
+	pods     getter
+	secrets  getter
+	issuer   token.TokenGenerator
+}
+
+func (r *TokenREST) New() runtime.Object {
+	return &authenticationapi.TokenRequest{}
+}
+
+func (r *TokenREST) Create(ctx genericapirequest.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, includeUninitialized bool) (runtime.Object, error) {
+	if err := createValidation(obj); err != nil {
+		return nil, err
+	}
+
+	out := obj.(*authenticationapi.TokenRequest)
+
+	svcacctObj, err := r.svcaccts.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	svcacct := svcacctObj.(*api.ServiceAccount)
+
+	var (
+		pod    *api.Pod
+		secret *api.Secret
+	)
+
+	if ref := out.Spec.BoundObjectRef; ref != nil {
+		var uid types.UID
+
+		gvk := schema.FromAPIVersionAndKind(ref.APIVersion, ref.Kind)
+		gk := gvk.GroupKind()
+		switch {
+		case gk == schema.GroupKind{Group: "", Kind: "Pod"}:
```

Comment:
Why not:
`case gk.Group == "" && gk.Kind == "Pod"`

Doesn't matter much either way, but I think you save the extra stack initialization of the new `GroupKind` object.